src/modules/Staff/StaffDetailsPage.tsx

```diff
@@ -0,0 +1,138 @@
+import { styled, useTheme } from '@mui/material/styles';
+import {
+  DetailsTile,
+  Header,
+  Navbar,
+  Paragraph,
+  StyledSkeleton,
+  ListPageSkeleton,
+  ErrorMessage,
+} from '@UG/libs/components';
+import { Academic } from '@UG/libs/types';
+import { useEffect, useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import { useNavigate, useParams } from 'react-router-dom';
+import { StateType } from 'src/store';
+import { useTranslation } from 'react-i18next';
+import { Box } from '@mui/material';
+import { useGetStaff } from './hooks';
+
+const Line = styled('div')`
+  width: 975px;
+  margin-bottom: 35px;
+  border-bottom: 1px solid ${({ theme }) => theme.palette.primary.dark};
+`;
+
+interface StateProps {
+  isLoading: boolean;
+  staffDetails: Academic | null;
+  errorMessage: string | null;
+}
+
+export const StaffDetailsPage = () => {
+  const theme = useTheme();
+  const { t } = useTranslation();
+  const navigate = useNavigate();
+  const { _id } = useParams();
+  const { getStaffDetails } = useGetStaff();
+
+  const { isLoading, staffDetails, errorMessage } = useSelector<StateType, StateProps>(state => ({
+    isLoading: state.staff.isLoading,
+    staffDetails: state.staff.staffDetails,
+    errorMessage: state.staff.error,
+  }));
+
+  useEffect(() => {
+    if (!_id) {
+      navigate('/error');
+    } else {
+      getStaffDetails(_id);
+    }
+  }, [_id, navigate, getStaffDetails]);
+
+  const academicPosts: JSX.Element[] = useMemo(() => {
+    if (!staffDetails || !staffDetails.content) {
+      return [];
+    }
+    return staffDetails.content?.posts.map((post, i) => (
+      <>
+        <Paragraph key={i} margin={'50px 0 50px 0'} fontSize={28} align={'center'}>
+          {post.position}
+        </Paragraph>
+        {post.faculty.map((f, index) => (
+          <DetailsTile key={index} width={975} marginTop={35} padding={'10px 40px'}>
+            <Paragraph margin={'25px'} color={theme.palette.primary.main} fontWeight={600}>
+              {f}
+            </Paragraph>
+          </DetailsTile>
+        ))}
+      </>
+    ));
+  }, [staffDetails, theme.palette.primary.main]);
+
+  //TODO change layout as soon as we get designs
+  if (!isLoading && errorMessage) {
+    return (
+      <>
+        <Header />
+        <ErrorMessage />
```

Comment:
`ErrorMessage` component should be given a props with the error message itself, your solution doesn't show the user which error occured